docs/ql-style-guide.md

```diff
@@ -1,417 +1,425 @@
-| *variable* | A parameter to a predicate, a field, a from variable, or a variable introduced by a *quantifier* or *aggregation*. |
+# QL Style Guide
+
+## Introduction
+
+This document describes how to format the QL code you contribute to this repository. It covers aspects such as layout, white-space, naming and documentation. Adhering to consistent standards makes code easier to read and maintain. Of course, these are only guidelines, and can be overridden as the need arises on a case-by-case basis. Where existing code deviates from these guidelines, prefer consistency with the surrounding code.
+
+Words in *italic* are defined in the [Glossary](#glossary).
+
+## Indentation
+1. *Always* use 2 spaces for indentation.
+1. *Always* indent:
+   - The *body* of a module, newtype, class or predicate
+   - The second and subsequent lines after you use a line break to split a long line
+   - The *body* of a `from`, `where` or `select` clause where it spans multiple lines
+   - The *body* of a *quantifier* that spans multiple lines
+
+
+### Examples
+
+```ql
+module Helpers {
+  /** ... */
+  class X ... {
+    /** ... */
+    int getNumberOfChildren () {
+      result = count(int child |
+        exists(this.getChild(child))
+      )
+    }
+  }
+}
+```
+
+```ql
+from Call c, string reason
+where isDeprecated(c, reason)
+select c, "This call to '$@' is deprecated because " + reason + ".",
+  c.getTarget(), c.getTarget().getName()
+```
+
+## Line breaks
+1. Use UNIX line endings.
+1. Lines *must not* exceed 100 characters.
+1. Long lines *should* be split with a line break, and the following lines *must* be indented one level until the next "regular" line break.
+1. There *should* be a single blank line:
+   - Between the file documentation and the first `import`
+   - Before each declaration, except for the first declaration in a *body*
+   - Before the `from`-`where`-`select` section in a query file
+1. *Avoid* two or more adjacent blank lines.
+1. There *must* be a new line after the *annotations* `cached`, `pragma`, `language` and `bindingset`. Other *annotations* do not have a new line.
+1. There *should not* be additional blank lines within a predicate.
+1. There *may* be a new line:
+   - Immediately after the `from`, `where` or `select` keywords in a query.
+   - Immediately after `if`, `then`, or `else` keywords. The `then` and `else` parts *should* be consistent.
+1. *Avoid* other line breaks in declarations, other than to break long lines.
+1. When operands of *binary operators* span two lines, the operator *should* be placed at the end of the first line.
+1. If the parameter list needs to be broken across multiple lines then there must *must* be a line break after the opening `(`, the parameter declarations indented one level, and the `) {` on its own line at the outer indentation.
```

Comment:
Double `must`

Reply:
Perhaps it would read better with an additional `*must* be` in the last part?
>  and the `) {` *must* be on its own line at the outer indentation.

Reply:
I initially thought at the inner indentation, but I agree with what you have written here. The "outer indentation" would be consistent with quantifiers and other parentheses.

Reply:
fixed